src/node.js

```diff
@@ -390,6 +391,61 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var pendingRejections = [];
```

Comment:
see #525. `for..of` is much slower than iterating manually, though we reused `IterationResult` objects